core/types.ml

```diff
@@ -2170,19 +2288,21 @@ struct
       | `Presence f -> "::Presence (" ^ presence bound_vars p f ^ ")"
 
   let tycon_spec bound_vars p =
+    let bound_vars tyvars =
+      List.fold_left
+        (fun bound_vars tyvar ->
+           TypeVarSet.add (var_of_quantifier tyvar) bound_vars)
+        bound_vars tyvars in
+
     function
       | `Alias (tyvars, body) ->
-          let bound_vars =
-            List.fold_left
-              (fun bound_vars tyvar ->
-                 TypeVarSet.add (var_of_quantifier tyvar) bound_vars)
-              bound_vars tyvars
-          in
-            begin
-              match tyvars with
-                | [] -> datatype bound_vars p body
-                | _ -> mapstrcat "," (quantifier p) tyvars ^"."^ datatype bound_vars p body
-            end
+          let bvs = bound_vars tyvars in
+          begin
```

Comment:
Redundant `begin ... end`?

Reply:
This one isn't redundant, as without the `begin ... end`, the `Mutual` and `Abstract` cases are parsed as being part of the inner `match`.